apps/itest/test/itest/batch_transactions_test.exs

```diff
@@ -0,0 +1,330 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule BatchTransactionsTests do
+  use Cabbage.Feature, async: true, file: "batch_transactions.feature"
+  @moduletag :transactions
+
+  require Logger
+
+  alias ChildChainAPI.Api.Transaction
+  alias ChildChainAPI.Connection, as: ChildChain
+  alias ChildChainAPI.Model.TransactionBatchSubmitBodySchema
+  alias ExPlasma.Transaction.Payment
+  alias Itest.Account
+  alias Itest.ApiModel.WatcherSecurityCriticalConfiguration
+  alias Itest.Client
+  alias Itest.Fee
+  alias Itest.Poller
+  alias Itest.Transactions.Currency
+  alias Itest.Transactions.Encoding
+
+  setup do
+    {:ok, _} = DebugEvents.start_link()
+
+    [{alice_address, alice_pkey}, {bob_address, bob_pkey}, {eve_address, eve_pkey}] = Account.take_accounts(3)
+
+    eth_fee =
+      Currency.ether()
+      |> Encoding.to_hex()
+      |> Fee.get_for_currency()
+      |> Map.get("amount")
+
+    %{
```

Comment:
you can replace it with `Enum.map()` since all structs are almost the same

Reply:
touche!